File: src/main/scala/fr/xebia/streams/video/RemoteProvider.scala
package fr.xebia.streams.video

sealed trait RemoteProvider {

  def width: Int

  def height: Int

  def uri: String
}

case class RPiCamWebInterface(host: String) extends RemoteProvider {

  val width = 512

  val height = 288

  override def uri: String = s"http://$host/html/cam_pic_new.php"

}

File: src/main/java/fr/xebia/streams/java/video/RemoteProvider.java
package fr.xebia.streams.java.video;

public class RemoteProvider {
    Integer width;
    Integer height;
    String url;

    public RemoteProvider() {
        this.width = 512;
        this.height = 288;
        this.url = "http://$host/html/cam_pic_new.php";
    }

    public Integer getWidth() {
        return width;
    }

    public void setWidth(Integer width) {
        this.width = width;
    }

    public Integer getHeight() {
        return height;
    }

    public void setHeight(Integer height) {
        this.height = height;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }
}
